packages/thirdweb/src/react/web/ui/Bridge/TransactionPayment.tsx

```diff
@@ -327,7 +327,11 @@ export function TransactionPayment({
           onClick={() => {
             if (transactionDataQuery.data?.tokenInfo) {
               onContinue(
-                transactionDataQuery.data.totalCost,
+                Math.max(
+                  0,
+                  Number(transactionDataQuery.data.totalCost) -
+                    Number(transactionDataQuery.data.userBalance),
```

Comment:
Might be easier to just fetch the balance separately so it doesn't break the main query if there's no account